lucille-core/src/main/java/com/kmwllc/lucille/core/fileHandler/JsonFileHandler.java

```diff
@@ -70,12 +91,28 @@ public Document next() {
 
         String line = it.next();
         try {
-          return Document.createFromJson(line, idUpdater);
+          if (idFields.isEmpty()) {
+            return Document.createFromJson(line, idUpdater);
+          } else {
+            ObjectNode node = (ObjectNode) mapper.readTree(line);
+            List<String> parts = new ArrayList<>(idFields.size());
+
+            for (String f : idFields) {
+              JsonNode v = node.get(f);
+              parts.add((v != null && !v.isNull()) ? v.asText() : "");
+            }
+
+            String rawId = (docIdFormat != null) ? String.format(docIdFormat, parts.toArray()) : String.join("_", parts);
+            node.put(Document.ID_FIELD, docIdPrefix + rawId);
```

Comment:
there's an edge case here where the json has a field called "id" but the user configures the filehandler to use a different field as the id

in this case, we're overwriting the "id" that's in the source json

this may be OK but we should document the behavior (in the class level javadocs for this file handler)